### PR TITLE
faster build by reduce by-product, reduce linking library and fix compile warning of std11

### PR DESCRIPTION
**Pull request by zhwesky2010**
1. Speed up the compilation, remove the Useless intermediates of paddle_fluid/paddle_fluid_origin/paddle_fluid_shared/memory/paddle_framwork , and change them to be library which does't cost time. As a result, the dependency relationship between targets has been adjusted, too.

2. Reduced some linked libs of testing executables. Currently, there are unnecessary libs are linked into testing executables.

3. Eliminates compilation warnings caused by std=c++11.

4. Some message warnings of cmake have been removed to avoid disturbing users.

5. Add tensorrt information to version.txt of inference lib.

It has been checked by **PR_Windows_CI_Inference**
http://ci.paddlepaddle.org/viewLog.html?buildId=262210&buildTypeId=PaddleWindows_PrWindowsCiInference_2&tab=buildResultsDiv&branch_PaddleWindows=pull%2F21521

### CODE REVIEW

File: cmake/inference_lib.cmake [hunk 1]
```diff
@@ -165,39 +165,30 @@ copy_part_of_thrid_party(inference_lib_dist ${FLUID_INFERENCE_INSTALL_DIR})
...
-    set(paddle_fluid_lib ${PADDLE_BINARY_DIR}/paddle/fluid/inference/${CMAKE_BUILD_TYPE}/libpaddle_fluid.*)
+    set(paddle_fluid_lib ${PADDLE_BINARY_DIR}/paddle/fluid/inference/${CMAKE_BUILD_TYPE}/*paddle_fluid.*)
 else(WIN32)
     set(paddle_fluid_lib ${PADDLE_BINARY_DIR}/paddle/fluid/inference/libpaddle_fluid.*)
 endif(WIN32)
 
 copy(inference_lib_dist
-        SRCS  ${src_dir}/inference/api/paddle_*.h ${paddle_fluid_lib}
+        SRCS  ${src_dir}/inference/api/paddle_*.h  ${paddle_fluid_lib}
         DSTS  ${FLUID_INFERENCE_INSTALL_DIR}/paddle/include ${FLUID_INFERENCE_INSTALL_DIR}/paddle/lib)
```
Comment (liupluswei) [inline cmake/inference_lib.cmake:184]: remove this change

File: cmake/generic.cmake [hunk 2]
```diff
@@ -226,6 +252,7 @@ function(merge_static_libs TARGET_NAME)
       # Get the file names of the libraries to be merged
       set(libfiles ${libfiles} $<TARGET_FILE:${lib}>)
     endforeach()
+    message("================${libfiles}")
```
Comment (liupluswei) [inline cmake/generic.cmake]: remove the log code

File: cmake/external/xxhash.cmake [hunk 3]
```diff
@@ -25,7 +25,7 @@ cache_third_party(extern_xxhash
     TAG           ${XXHASH_TAG}
     DIR           ${XXHASH_PREFIX_DIR})
 
-IF(WITH_STATIC_LIB)
+IF(XXHASH_STATIC_LIB)
```
Comment (liupluswei) [inline cmake/external/xxhash.cmake]: remove this

Comment (zhwesky2010) [reply cmake/external/xxhash.cmake]: Done

File: cmake/inference_lib.cmake (hunk 1 above)
Comment (zhwesky2010) [reply cmake/inference_lib.cmake:184]: Done

File: cmake/generic.cmake (hunk 2 above)
Comment (zhwesky2010) [reply cmake/generic.cmake]: Done

Comment (zhwesky2010) [reply cmake/generic.cmake]: Done

**Pull request closed without merging**